src/sentry/integrations/slack/utils.py

```diff
@@ -356,27 +361,44 @@ def get_channel_id(organization, integration_id, name):
     # Look for channel ID
     payload = dict(token_payload, **{"exclude_archived": False, "exclude_members": True})
 
+    # XXX(meredith): For large accounts that have many, many channels it's
+    # possible for us to timeout while attempting to paginate through to find the channel id
+    # This means some users are unable to create/update alert rules. To avoid this, we attempt
+    # to find the channel id asynchronously if it takes longer than a certain amount of time,
+    # which I have set here - arbitrarily - to 10 seconds.
+    timeout = time.time() + 10
     session = http.build_session()
     for list_type, result_name, prefix in LIST_TYPES:
-        # Slack limits the response of `<list_type>.list` to 1000 channels, paginate if
-        # needed
         cursor = ""
         while cursor is not None:
+
+            if time.time() > timeout:
+                return (prefix, None, True)
+
             items = session.get(
                 "https://slack.com/api/%s.list" % list_type,
-                params=dict(payload, **{"cursor": cursor}),
+                # TODO(meredith): change this to 1000
+                # Slack limits the response of `<list_type>.list` to 1000 channels
+                params=dict(payload, **{"cursor": cursor, "limit": 1}),
```

Comment:
@MeredithAnya why are we doing limit `1` instead of using `1000`? Won't we have to make potentially thousands of API calls if the limit is `1`?

Reply:
yeah was using 1 for testing - can change that now